### PR TITLE
Replace testify/assert with gotestyourself/assert

### PR DESCRIPTION
**Pull request by dnephin**
[gotestyourself/assert](https://godoc.org/github.com/gotestyourself/gotestyourself/assert) is being used by `docker/cli` and `container/containerd`. The `gotestyourself` package is already being used by `moby/moby` for other utilities, so using it for `assert` removes two dependencies (testify and go-spew).

`gotestyourself/assert` fixes many of the [issues with testify](https://gist.github.com/dnephin/1393e56cb4ed713221302f7949c53012). It uses https://godoc.org/github.com/google/go-cmp/cmp which allows for a lot more flexibility when comparing nested structures. `cmpJSONMessageOpt`, and ` cmpServiceOpts` in this PR are a couple examples.
[TestVolumesInspect](https://github.com/moby/moby/blob/f3a3778808dae0da82f2acf300875b5da931c000/integration/volume/volume_test.go#L103-L107) is an example of where this support could be used.

The "Automated Migrate" commit was automated using [assert/cmd/gty-migrate-from-testify](https://github.com/gotestyourself/gotestyourself/tree/master/assert/cmd/gty-migrate-from-testify) which parses and writes AST. The behaviour of the old test suite is preserved, so there are lots of calls to `assert.Check()` (which is the equivalent of `testify/assert`, using `t.Fail`). These can be cleaned up in a follow up PR to use the convenience functions in `assert` (which use `t.FailNow`).

### CODE REVIEW

File: pkg/term/term_linux_test.go
Comment (dnephin) [inline pkg/term/term_linux_test.go]: This is a great example of one of the many problems with testify. These asserts did nothing because `int` can never be nil, so the assert always passes, regardless of the value.  

Passing any non-nillable type to `Nil/NotNil` should always fail.

**Comment by dnephin** [comment]
powerpc build failure was timeout, and windows failures look like flakes, so I think this is effectively green and ready for review.

**Comment by cpuguy83** [comment]
```
02:38:13 integration/network/service_test.go:49:2: undefined: require
02:38:13 integration/network/service_test.go:52:2: undefined: require
02:38:13 integration/network/service_test.go:67:2: undefined: require
02:38:13 integration/network/service_test.go:98:2: undefined: require
02:38:13 integration/network/service_test.go:118:2: undefined: require
02:38:13 integration/network/service_test.go:119:2: undefined: require
02:38:13 integration/network/service_test.go:120:2: undefined: require
02:38:13 integration/network/service_test.go:122:2: undefined: require
```

**Comment by thaJeztah** [comment]
Discussed this in the maintainers meeting, and generally we seem to be ok with changing to this framework

**Comment by dnephin** [comment]
Other than many flakes, this should be green now.

**Review by cpuguy83 — APPROVED** [review]
LGTM assuming everything is green.

**Comment by cpuguy83** [comment]
Z error
```
08:13:45 Step 97/97 : COPY . /go/src/github.com/docker/docker
08:13:47  ---> 87ec6fa5c2f1
08:13:47 Removing intermediate container 808b6d494192
08:13:48 Successfully built 87ec6fa5c2f1
08:13:48 Successfully tagged docker-s390x:1298382
08:13:50 can't load package: package github.com/stretchr/testify/assert: cannot find package "github.com/stretchr/testify/assert" in any of:
08:13:50 	/usr/local/go/src/github.com/stretchr/testify/assert (from $GOROOT)
08:13:50 	/go/src/github.com/stretchr/testify/assert (from $GOPATH)
08:13:50 can't load package: package github.com/stretchr/testify/require: cannot find package "github.com/stretchr/testify/require" in any of:
08:13:50 	/usr/local/go/src/github.com/stretchr/testify/require (from $GOROOT)
08:13:50 	/go/src/github.com/stretchr/testify/require (from $GOPATH)
08:13:51 Build step 'Execute shell' marked build as failure
```

**Comment by vdemeester** [comment]
arf need another rebase :sweat:

**Comment by dnephin** [comment]
Rebased, and it seems like everything is building correctly.

**Comment by dnephin** [comment]
There was a TearDown flake on z, otherwise this is green.

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Pull request merged**
